suppression-backup-service/exporter/exporter.go

```diff
@@ -0,0 +1,223 @@
+package exporter
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+	suppression "github.com/rudderlabs/rudder-server/enterprise/suppress-user"
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/suppression-backup-service/model"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+// ExporterOpt represents a configuration option for the exporter
+type ExporterOpt func(*Exporter)
+
+func Export(repo suppression.Repository, file model.File) error {
+	// export initially to a temp file
+	tempFile, err := os.CreateTemp("", "tmp-export")
+	if err != nil {
+		return fmt.Errorf("error creating temp file: %w", err)
+	}
+	defer func() {
+		_ = tempFile.Close()
+		_ = os.Remove(tempFile.Name())
+	}()
+
+	// export to temp file
+	err = repo.Backup(tempFile)
+	if err != nil {
+		return fmt.Errorf("error exporting suppression list: %w", err)
+	}
+
+	// move temp file to final destination
+	file.Mu.Lock()
+	defer file.Mu.Unlock()
+	err = os.Rename(tempFile.Name(), file.Path)
+	if err != nil {
+		return fmt.Errorf("error moving temp file to final destination: %w", err)
+	}
+
+	return nil
+}
+
+func newBadgerDBInstance(baseDir string, pkgLogger logger.Logger) (suppression.Repository, error) {
+	_ = os.RemoveAll(baseDir)
+	err := os.MkdirAll(baseDir, os.ModePerm)
+	if err != nil {
+		return nil, fmt.Errorf("could not create tmp dir: %w", err)
+	}
+	repository, err := suppression.NewBadgerRepository(
+		baseDir,
+		pkgLogger)
+	if err != nil {
+		return nil, fmt.Errorf("could not create badger repository: %w", err)
+	}
+	return repository, nil
+}
+
+type Exporter struct {
+	id             identity.Identifier
+	file           model.File
+	log            logger.Logger
+	pollIntervalFn func() time.Duration
+}
```

Comment:
**minor:** this struct appears to be redundant, i.e. the struct's fields could be moved to the respective functions:
- `FullExporterLoop`
- `LatestExporterLoop`

This way we could spare the `Exporter` struct, `ExporterOpt` type, `WithPollIntervalFn` & `NewExporter` functions.

Reply:
`WithPollIntervalFn` helps in making loop testable.

Reply:
by passing the same as a parameter to the function(s)
> the struct's fields could be moved to the respective functions